unit_test/batched/Test_Batched_BatchedGemm.hpp

```diff
@@ -117,6 +119,88 @@ void impl_test_batched_gemm(const int N, const int matAdim1, const int matAdim2,
+    cublasHandle_t cublas_handle;
+    BatchedGemmHandle batchedGemmHandleCublas(cublas_handle,
+                                              GemmTplAlgos::CUBLAS, 0, 0);
+    ASSERT_EQ(&cublas_handle, batchedGemmHandleCublas.get_tpl_params());
+    ASSERT_EQ(batchedGemmHandleCublas.get_kernel_algo_type(),
+              (int)GemmTplAlgos::CUBLAS);
+    ASSERT_EQ(batchedGemmHandleCublas.teamSz, 0);
+    ASSERT_EQ(batchedGemmHandleCublas.vecLen, 0);
+#endif
+
+#if defined(KOKKOSKERNELS_ENABLE_TPL_MAGMA)
+    magma_queue_t magma_queue;
+    BatchedGemmHandle batchedGemmHandleMagma(magma_queue, GemmTplAlgos::MAGMA,
+                                             0, 0);
+    ASSERT_EQ(&magma_queue, batchedGemmHandleMagma.get_tpl_params());
+    ASSERT_EQ(batchedGemmHandleMagma.get_kernel_algo_type(),
+              (int)GemmTplAlgos::MAGMA);
+    ASSERT_EQ(batchedGemmHandleMagma.teamSz, 0);
+    ASSERT_EQ(batchedGemmHandleMagma.vecLen, 0);
+#endif
+  }
+
+  for (int algo_type = BaseHeuristicAlgos::SQUARE;
+       algo_type < GemmKokkosBatchedAlgos::N; ++algo_type) {
+    {
+      BatchedGemmHandle batchedGemmHandle(algo_type);
+
+      ASSERT_EQ(batchedGemmHandle.get_kernel_algo_type(), algo_type);
+
+      if (algo_type == BaseKokkosBatchedAlgos::KK_SERIAL) {
+        // Invoke 4 times to ensure we cover all paths for alpha and beta
+        impl_test_batched_gemm_handle<DeviceType, ViewType, ScalarType,
+                                      ParamTagType>(
+            &batchedGemmHandle, N, matAdim1, matAdim2, matBdim1, matBdim2,
+            matCdim1, matCdim2, 0.0, 0.0);
+        impl_test_batched_gemm_handle<DeviceType, ViewType, ScalarType,
+                                      ParamTagType>(
+            &batchedGemmHandle, N, matAdim1, matAdim2, matBdim1, matBdim2,
+            matCdim1, matCdim2, 1.0, 0.0);
+        impl_test_batched_gemm_handle<DeviceType, ViewType, ScalarType,
+                                      ParamTagType>(
+            &batchedGemmHandle, N, matAdim1, matAdim2, matBdim1, matBdim2,
+            matCdim1, matCdim2, 0.0, 1.0);
+        impl_test_batched_gemm_handle<DeviceType, ViewType, ScalarType,
+                                      ParamTagType>(
+            &batchedGemmHandle, N, matAdim1, matAdim2, matBdim1, matBdim2,
+            matCdim1, matCdim2, 1.5, 3.0);
+      } else {
+        try {
+          // Allocate these views to invoke BatchedGemm with an unsupported algo
+          // type
+          ViewType a_actual("a_actual", N, matAdim1, matAdim2);
+          ViewType b_actual("b_actual", N, matBdim1, matBdim2);
+          ViewType c_actual("c_actual", N, matCdim1, matCdim2);
+          using ta = typename ParamTagType::transA;
+          using tb = typename ParamTagType::transB;
+          using bl = typename ParamTagType::batchLayout;
+          BatchedGemm<ta, tb, bl>(&batchedGemmHandle, 0.34, a_actual, b_actual,
+                                  0.43, c_actual);
+          FAIL();
```

Comment:
It looks to me that only ```KK_SERIAL``` is being tested but you have the FOR loop for all algorithms. Will the other tests fail? Am I missing the logic here?

Reply:
No, the other tests will not fail. When we extend the `BatchedGemm` interface support past `KK_SERIAL`, the other tests will fail. Currently, we are testing that the `BatchedGemm` interface correctly throws a runtime error for unsupported algorithm types. See the comment at https://github.com/kokkos/kokkos-kernels/pull/1054/files/cc5e1246ec4905c93e5c5ed0ee682b596eccba1c#diff-c549a249e74df38be0722b9ba309e85b0880584a9be326049a3c81a86307c74bR186.

Reply:
Just to understand it more: When a not-KK_SERIAL seleted, what does ```BatchedGemm<ta, tb, bl>(&batchedGemmHandle, 0.34, a_actual, b_actual, 0.43, c_actual);``` do? I think it does nothing and returns to the main code and then FAIL() is called? which means the test fails

Reply:
Get it. I thought it does not throw a runtime error for not-KK_SERIAL algorithm selection.